tests/virt_autotest/switch_version_and_reload_needle.pm

```diff
@@ -0,0 +1,29 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This test does not change the VERSION and only reload needles.
+#          It needs to be wrapped in test because
+#          set_var with reload_needles option needs to be run in test files rather than main.pm,
+#          It does not change VERSION together with reload_needles, because
+#          VERSION needs to be changed in main.pm, so as to load different tests accordingly
+# Maintainer: xlai@suse.com
+
+
+use strict;
+use warnings;
+use base "virt_autotest_base";
+use testapi;
+
+sub run {
+    #VERSION needs to be changed in main.pm, so as to load different tests accordingly
+    set_var('VERSION', get_var('VERSION'), reload_needles => 1);
```

Comment:
that's a tricky one - the main.pm will leave one VERSION behind, so what does change the VERSION variable during test runs if not this one here?

Reply:
Reason is given in the beginning of this test file as 

```
+# Summary: This test does not change the VERSION and only reload needles.
+#          It needs to be wrapped in test because
+#          set_var with reload_needles option needs to be run in test files rather than main.pm,
+#          It does not change VERSION together with reload_needles, because
+#          VERSION needs to be changed in main.pm, so as to load different tests accordingly
```

Reply:
Maybe the test file leads to misunderstanding. What it does is to trigger needle_reload only. And the real change variable action needs to be done previously in main.pm to let test loading correctly for installation which has whether or not load actions via sle_version checking.

So although this api is designed to do set var and reload needle together, however in real usage i have to split them.

Reply:
So wouldn't it be more obvious to call

```perl
set_var('VERSION', get_required_var('VERSION_TO_INSTALL'), reload_needles => 1);
```

and rename the test module accordingly? Then of course you need to create another test module to switch the version again, e.g. for migration after installation but you need unique names anyway because openQA does not handle the test module with the same name scheduled again well anyway (os-autoinst does, openQA not. So could be very confusing for people looking at the webui).